sphinx/content/ch1.md

```diff
@@ -0,0 +1,1249 @@
+in the same order as in our count data.
+This is where pandas indexing comes in handy!
+We can get the intersection of the gene names from our two sources of data
+and use these to index both datasets, ensuring they have the same genes in the same order.
+
+```python
+# Subset gene info to match the count data
+matched_index = pd.Index.intersection(data_table.index, gene_info.index)
+```
+
+Now let's use the intersection of the gene names to index our count data.
+
+```python
+# 2D ndarray containing expression counts for each gene in each individual
+counts = np.asarray(data_table.loc[matched_index], dtype=int)
+
+gene_names = np.array(matched_index)
+
+# Check how many genes and individuals were measured
+print(f'{counts.shape[0]} genes measured in {counts.shape[1]} individuals.')
+```
+
+And our gene lengths.
+
+```python
+# 1D ndarray containing the lengths of each gene
+gene_lengths = np.asarray(gene_info.loc[matched_index]['GeneLength'],
+                          dtype=int)
+```
+
+And let's check the dimensions of our objects.
+
+```python
+print(counts.shape)
+print(gene_lengths.shape)
+```
+
+As expected, they now match up nicely!
+
+## Normalization
+
+Real world data contains all kinds of measurement artifacts.
+Before doing any kind of analysis with it, it is important to take a look at
+it to determine whether some normalization is warranted.
+For example, measurements with digital thermometers may systematically vary from
+those taken with mercury thermometers and read out by a human.
+Thus, comparing samples often requires some kind of data wrangling to bring
+every measurement to a common scale.
+
+In our case, we want to make sure that any differences we uncover correspond to
+real biological differences, and not to technical artifact.
+We will consider two levels of normalization often applied jointly to gene
+expression dataset: normalization between samples (columns) and normalization
+between genes (rows).
+
+### Between Samples
+
+For example, the number of counts for each individual can vary substantially in RNAseq experiments.
+Let's take a look at the distribution of expression counts over all the genes.
+First, we will sum the columns to get the total counts of expression of all genes for each individual, so we can just look at the variation between individuals.
+To visualize the distribution of total counts, we will use kernel density
+estimation (KDE), a technique commonly used to smooth out histograms because it
+gives a clearer picture of the underlying distribution.
+
+Before we start, we have to do some plotting setup (which we will do in every
+chapter). See "A quick note on plotting" for details about each line of the following code.
+
+```python
+# Make all plots appear inline in the Jupyter notebook from now onwards
+%matplotlib inline
+# Use our own style file for the plots
+import matplotlib.pyplot as plt
+plt.style.use('style/elegant.mplstyle')
+```
+
+
+> **A Quick Note on Plotting {.callout}**
```

Comment:
Is this callout syntax still acceptable?

Reply:
One nice thing is that you can use `admonition` directives instead, which include all kinds of subclasses like `hint`, `warning`, `tip`, etc. I used a few of them more liberally in ch2.md. So the syntax does change ({.callout} is not part of the commonmark standard), but in my opinion the directives are nicer for callouts.

OTOH I haven't figured out footnotes yet, or a nice way to simulate footnote-like behavior with the given set of directives.